libminifi/include/core/state/nodes/AgentInformation.h

```diff
@@ -425,77 +427,123 @@ class AgentStatus : public StateMonitorNode {
+      queuesize.name = "size";
+      queuesize.value = repo.second->getRepoSize();
 
-        SerializedResponseNode isFull;
-        isFull.name = "full";
-        isFull.value = repo.second->isFull();
+      SerializedResponseNode isRunning;
+      isRunning.name = "running";
+      isRunning.value = repo.second->isRunning();
 
-        repoNode.children.push_back(queuesize);
-        repoNode.children.push_back(isRunning);
-        repoNode.children.push_back(isFull);
-        repositories.children.push_back(repoNode);
-      }
-      serialized.push_back(repositories);
+      SerializedResponseNode isFull;
+      isFull.name = "full";
+      isFull.value = repo.second->isFull();
+
+      repoNode.children.push_back(queuesize);
+      repoNode.children.push_back(isRunning);
+      repoNode.children.push_back(isFull);
+      repositories.children.push_back(repoNode);
     }
+    return repositories;
+  }
 
-    serialized.push_back(uptime);
+  SerializedResponseNode serializeUptime() {
+    SerializedResponseNode uptime;
 
+    uptime.name = "uptime";
     if (nullptr != monitor_) {
+      uptime.value = monitor_->getUptime();
+    } else {
+      uptime.value = "0";
+    }
+
+    return uptime;
+  }
+
+  SerializedResponseNode serializeComponents() {
+    SerializedResponseNode components_node(false);
+    components_node.name = "components";
+    if (monitor_ != nullptr) {
       auto components = monitor_->getAllComponents();
-      SerializedResponseNode componentsNode(false);
-      componentsNode.name = "components";
 
       for (auto component : components) {
-        SerializedResponseNode componentNode(false);
-        componentNode.name = component->getComponentName();
+        SerializedResponseNode component_node(false);
+        component_node.name = component->getComponentName();
 
-        SerializedResponseNode uuidNode;
-        uuidNode.name = "uuid";
-        uuidNode.value = std::string{component->getComponentUUID().to_string()};
+        SerializedResponseNode uuid_node;
+        uuid_node.name = "uuid";
+        uuid_node.value = std::string{component->getComponentUUID().to_string()};
 
-        SerializedResponseNode componentStatusNode;
-        componentStatusNode.name = "running";
-        componentStatusNode.value = component->isRunning();
+        SerializedResponseNode component_status_node;
+        component_status_node.name = "running";
+        component_status_node.value = component->isRunning();
 
-        componentNode.children.push_back(componentStatusNode);
-        componentNode.children.push_back(uuidNode);
-        componentsNode.children.push_back(componentNode);
+        component_node.children.push_back(component_status_node);
+        component_node.children.push_back(uuid_node);
+        components_node.children.push_back(component_node);
       }
-      serialized.push_back(componentsNode);
     }
+    return components_node;
+  }
+
+  SerializedResponseNode serializeAgentMemoryUsage() {
```

Comment:
This func could be const

Reply:
done [d2832cf](https://github.com/apache/nifi-minifi-cpp/pull/1032/commits/d2832cfcb3afa837e30e7bc95f496f0754d7603f)